src/const_assert.rs

```diff
@@ -52,7 +52,10 @@
 macro_rules! const_assert {
     ($x:expr $(,)?) => {
         #[allow(unknown_lints, eq_op)]
-        const _: [(); 0 - !{ const ASSERT: bool = $x; ASSERT } as usize] = [];
+        const _: fn() = ||{
+            const ASSERT: bool = $x;
+            let _: $crate::True = $crate::to_bool!($x);
```

Comment:
Not that you should do this (because obscurity), but I think pattern matching allows `let $crate::True = $crate::to_bool!($x);` to compile also. 😬